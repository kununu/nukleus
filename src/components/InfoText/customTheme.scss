.infoText {
  color: blue;
  font-size: 12px;
  margin: 0 0 10px;
  
  &.inline {
    margin-left: 127px;
  }
}

