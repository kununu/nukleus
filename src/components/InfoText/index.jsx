import React from 'react';
import PropTypes from 'prop-types';

import styles from './index.scss';

export default function InfoText ({text, infoStyle}) {
  return (
    <div className={`${styles.infoText} ${styles[infoStyle]}`}>
      {text}
    </div>
  );
}

InfoText.propTypes = {
  infoStyle: PropTypes.oneOf(['inline']),
  text: PropTypes.string.isRequired,
};

InfoText.defaultProps = {infoStyle: 'inline'};
