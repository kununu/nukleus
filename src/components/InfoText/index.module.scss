@import '~@kununu/kununu-theme-v2/scss/base/variables';

.infoText {
  color: $gray-base-40;
  font-size: $font-size-smaller;
  font-style: italic;
  margin: 0 0 10px;

  @media (max-width: $screen-xs-min) {
    margin-left: 0;
  }
}

.infoText.inline {
  margin-left: 127px;
}

