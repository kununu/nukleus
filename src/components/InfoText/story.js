import React from 'react';
import {storiesOf} from '@storybook/react';
import {defaultOptions} from 'defaultOptions';
import {withOptions} from '@storybook/addon-options';
import {withInfo} from '@storybook/addon-info';
import {withKnobs, text} from '@storybook/addon-knobs/react';

import TextField from '../TextField';

import InfoText from './index';

storiesOf('InfoText', module)
  .addDecorator(withOptions({...defaultOptions}))
  .addDecorator(withKnobs)
  .addDecorator((story, context) => withInfo('The info text component!')(story)(context))
  .add('basics', () => (
    <div>
      <TextField
        id="text-field"
        label="TextField"
        name="text-field" />

      <div>
        <InfoText
          text={text('text', 'I am an info text.')} />
      </div>
    </div>
  ));
