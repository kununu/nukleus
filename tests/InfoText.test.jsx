import React from 'react';
import renderer from 'react-test-renderer';
import InfoText from 'InfoText'; // eslint-disable-line import/no-unresolved, import/extensions, import/no-extraneous-dependencies

test('Renders InfoText without crashing', () => {
  const component = renderer.create(<InfoText
    text="test" />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
