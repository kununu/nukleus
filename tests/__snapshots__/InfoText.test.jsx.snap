// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders InfoText without crashing 1`] = `
<div
  className="infoText inline"
>
  test
</div>
`;
